packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -311,6 +317,22 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
     }
   }
 
+  void _handTapMove(TapMoveDetails event) {
+    final double distance = switch (defaultTargetPlatform) {
```

Comment:
I'm a bit troubled by the duplicate switch statement here and in the unit test. What do you think if we extract this switch statement as a function and replace the `kCupertinoButtonTapMoveOpacityChangeDistance` constant?